Add option to configure machine config pools (#50)
appuio/component-openshift4-nodes: docs/modules/ROOT/pages/references/parameters.adoc

Values given here will be merged with precedence with the defaults configured in <<defaultSpec>>.
The values can be everything that's accepted in the `spec` field of a `machinesets.machine.openshift.io` object.

== `machineConfigPools`

[horizontal]
type:: dictionary

A dictionary of machine config pools to create on the cluster.
The resulting `MachineConfigPool` object will have the key prefixed with `x-` as the name and the value is merged into the resource.

Apart from the machine config pool, this dictionary can manage related resources through the following fields, that won't be added to the `MachineConfigPool` object.

* The **kubelet** field can contain the spec of a `KubeletConfig` resource.
The machine config pool key will be used as the name and the `machineConfigPoolSelector` will be set automatically.
* The **containerRuntime** field can contain the spec of a `ContainerRuntimeConfig` resource.

> The machine config pool key will be used as the name and the `machineConfigPoolSelector` will be set automatically.

Reviewer: I don't like having multiple ways to specify containerRuntimeConfigs and kubeletConfigs, but I guess it's fine for now.

If we keep the implementation as-is for now, I'd like to have a follow-up issue to mark the top-level `kubeletConfigs` and `containerRuntimeConfigs` fields as deprecated so that we can remove them at some point.

Reviewer: We probably shouldn't just remove the old parameters without a deprecation period, since I assume that there's a number of configurations which actually make use of them. I'd much prefer spending some extra time in a follow-up issue / PR to make the migration as painless as possible on the config side (regardless of whether a maintenance window is required for this PR).

Author: Opened https://github.com/appuio/component-openshift4-nodes/issues/51 as a follow up

* The **machineConfigs** accepts a key-value dict where the values are the `spec` of `MachineConfig` resources.
The resulting `MachineConfig` objects will have the keys, prefixed with `99x-` and the machine config pool key, as the name and the values are used as the `spec`.
The objects will automatically be labeled so that the machine config pool will pick up the config.

> By default the component creates `app`, `infra`, and `storage` machine config pools, which are an extension of the `worker` pool.

Reviewer: ```suggestion
By default the component creates `app`, `infra`, and `storage` machine config pools, each of which extends the `worker` pool.
```

      containerRuntime: <2>
        containerRuntimeConfig:
          pidsLimit: 1337
      machineConfigs:
        testfile: <3>
          config:
            ignition:
              version: 3.2.0
            storage:
              files:
              - contents:
                  source: data:,custom
                filesystem: root
                mode: 0644
                path: /etc/customtest
        testfile2: <4>
          config:
            ignition:
              version: 3.2.0
            storage:
              files:
              - contents:
                  source: data:,custom
                filesystem: root
                mode: 0644
                path: /etc/customtest2
----
<1> Results in a `KubeletConfig` object called `app`
<2> Results in a `ContainerRuntimeConfig` object called `app`
<3> Results in a `MachineConfig` object called `99x-app-testfile`
<4> Results in a `MachineConfig` object called `99x-app-testfile2`

[NOTE]
====
Machine config pool names are prefixed with `x-` as in some cases configurations are applied ordered by their name and we want the config for `x-app` to be applied after the default worker config of the `worker` pool.
====

> [NOTE]

Reviewer: IMO this is worth to be marked "IMPORTANT"

```suggestion
[IMPORTANT]
```
